src/ruby/puppet-server-lib/puppet/server/certificate.rb

```diff
@@ -15,6 +16,7 @@ def initialize(java_cert)
     end
 
     @java_cert = java_cert
+    @oid_defns = Puppet::Server::Oids.load_custom_oid_file(Puppet[:trusted_oid_mapping_file]) + Puppet::SSL::Oids::PUPPET_OIDS
```

Comment:
Another problem I ran into is if the file pointed to by the `trusted_oid_mapping_file` setting does not exist, I get an error every time I make a request:

```
2016-03-29 17:23:42,249 ERROR [qtp1610975141-41] [puppet-server] Puppet Server Error: undefined method `+' for nil:NilClass
{noformat}

I think the implementation needs to handle just using {{Puppet::SSL::Oids::PUPPET_OIDS}} as a base list but add on the ones from the `trusted_oid_mapping_file` if it exists.